packages/client-interaction/runtime/directive.ts

```diff
@@ -9,8 +9,20 @@ const firstInteraction = new Promise<void>(resolve => {
     }
 })
 
-export default (async load => {
+export default (async (load, opts) => {
     await firstInteraction
-    const hydrate = await load()
-    await hydrate()
+    const hy = async () => {
+        const hydrate = await load()
+        await hydrate()
+    }
+    const arrOpts = Array.isArray(opts.value) ? opts.value : [opts.value]
+    if (arrOpts.includes('idle')) {
+        if (typeof window.requestIdleCallback === 'function') {
+            window.requestIdleCallback(hy)
+            return
+        }
```

Comment:
If I understand correctly, the idea is to delay the loading the page is also idle. Could you elaborate on when that would be preferable?

Reply:
Hey, sorry for the lack of explanation 🙃

The idle option is to wait the next browser idle period to call component hydration.
When having many hydrated components you probably have some `client:idle` and probably believe it'll be hydrated before `client:interaction` (and after `client:load`), but it's not necessarily true cause first interaction may occur before first idle.

Also, if using view transitions `client:interaction` will act like `client:load` on navigation, sometimes even before `client:load`, in concurrency with (and thus delaying) more important components.

Reply:
Thanks, that makes a lot of sense!

The big consideration now is the size of the directive. For performance reasons, the code for the directive is inlined into html, which means if there are 5 islands using `client:interaction`, there will be 5 copies of this code in the initial response.

Do you think it makes sense for it to be another directive (with fewer lines of code) instead of an option for `client:interaction`?

Reply:
Are you sure directive code is replicated each island? 🤔 
I didn't know that, if it is true I will have to review a custom directive that I use, but from what I'm testing the code only appears once even with more than one component with the directive on the page

Reply:
Take a look https://demo.ecomplus.app/p-blusa-charlotte-ml-cinza-mid
In the output HTML code we have two `client="context"`, but only one `addEventListener("storefront:apiContext"`

Reply:
An I missing something?

Reply:
You're right! It is inlined but included at most once per page.